clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -8600,8 +8600,23 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
   TranslatorArgs.push_back("-o");
   TranslatorArgs.push_back(Output.getFilename());
   if (getToolChain().getTriple().isSYCLDeviceEnvironment()) {
+    bool isFPGAHardware = false;
+    if (getToolChain().getTriple().getSubArch() ==
+        llvm::Triple::SPIRSubArch_fpga) {
+      for (auto *A : TCArgs) {
+        if (A->getOption().matches(options::OPT_Xs_separate) ||
+            A->getOption().matches(options::OPT_Xs)) {
+          StringRef ArgString(A->getValue());
+          if (ArgString == "hardware" || ArgString == "simulation")
+            isFPGAHardware = true;
+        }
+      }
+    }
     TranslatorArgs.push_back("-spirv-max-version=1.3");
-    TranslatorArgs.push_back("-spirv-debug-info-version=ocl-100");
+    if (!isFPGAHardware)
```

Comment:
We should be able to use `C.getDriver().isFPGAEmulationMode()` here instead of using the logic against `-Xs` above.

Reply:
Done, thanks!